osu.Game/Beatmaps/BeatmapDifficultyCache.cs

```diff
@@ -136,7 +136,7 @@ protected override void LoadComplete()
             var localRulesetInfo = rulesetInfo as RulesetInfo;
 
             // Difficulty can only be computed if the beatmap and ruleset are locally available.
-            if (localBeatmapInfo?.IsManaged != true || localRulesetInfo == null)
+            if (localBeatmapInfo == null || localRulesetInfo == null)
```

Comment:
This change on `localBeatmapInfo` is also not necessarily correct. For instance, if you open the [playlists results screen](https://github.com/ppy/osu/blob/4ca2822d409ff8b4b17213f9ef1c2916d8fdd706/osu.Game/Screens/OnlinePlay/Playlists/PlaylistsResultsScreen.cs#L183) for a beatmap that is not happily available, a dummy beatmap info will happily make its way here and pass this check (which it would fail previously):

![2022-01-15-173420_1516x73_scrot](https://user-images.githubusercontent.com/20418176/149629762-57de61ba-e5e1-46a8-bef3-868c8a5f922f.png)

This causes the score panels on said playlists results screen to show 0 stars.

Reply:
Not 100% on this one. I don't think this is the underlying cause of that issue (see https://github.com/ppy/osu/issues/16480).